modules/plugin/jp2k/src/main/java/org/geotools/coverageio/jp2k/ReadType.java

```diff
@@ -92,90 +82,25 @@ RenderedImage read(
     				LOGGER.log(Level.WARNING,"Unable to compute source area for file "
     						+ rasterFile.getAbsolutePath(),e);	
     			return null;
-    		} finally {
-    			//close everything
-    			try {
-    				// reader
-    				reader.dispose();
-    			} catch (Throwable t) {
-    				// swallow the exception, we are just trying to close as much stuff as possible
-    			} 
-    			
-    			try {
-    				// instream
-    				inStream.close();
-    			} catch (Throwable t) {
-    				// swallow the exception, we are just trying to close as much stuff as possible
-    			} 							
-    		}        		
+    		}
     	}
     }, 
     
     
     JAI_IMAGEREAD{
     	@Override
     	RenderedImage read(
-    			final ImageReadParam readP,
-    			final int imageIndex, 
-    			final File rasterFile,
-    			final Rectangle readDimension,
-    			final Dimension tileDimension,
-    			final ImageReaderSpi spi
-    			) throws IOException{
+                final ImageReadParam readP,
+                final int imageIndex,
+                final File rasterFile,
+                final Rectangle readDimension,
+                final Dimension tileDimension,
+                final ImageReaderSpi spi,
+                ImageReader reader) throws IOException{
     		
-      		///
-    		// Using ImageReader to load the data directly
-    		//
-    		ImageInputStream inStream=null;
-    		ImageReader reader=null;
-    		try{
-    			//get stream
-    			inStream = Utils.getInputStream(rasterFile);
-    			if(inStream==null)
-    				return null;
-    			// get a reader
-    			reader=spi.createReaderInstance();
-    			if(reader==null)
-    			{
-    				if (LOGGER.isLoggable(Level.WARNING))
-    					LOGGER.warning("Unable to get reader for file "
-    							+ rasterFile.getAbsolutePath());
-    				return null;
-    			}
-    			
-    			inStream.reset();
-    			reader.setInput(inStream);
+			if(CoverageUtilities.checkEmptySourceRegion(readP, readDimension))
+				return null;
```

Comment:
And curly braces around all if clauses. Local reformats may be necessary to fix tabs and for local consistency.